test/e2e/kubectl/kubectl.go

```diff
@@ -1320,46 +1321,47 @@ metadata:
 				return strings.Split(strings.TrimRight(out, "\n"), "\n")
 			}
 
-			ginkgo.By("Waiting for Redis master to start.")
-			waitForOrFailWithDebug(1)
-			forEachPod(func(pod v1.Pod) {
-				ginkgo.By("checking for a matching strings")
-				_, err := framework.LookForStringInLog(ns, pod.Name, containerName, "Ready to accept connections", framework.PodStartTimeout)
-				framework.ExpectNoError(err)
+			ginkgo.By("Waiting for log generator to start.")
+			if !e2epod.CheckPodsRunningReadyOrSucceeded(c, ns, []string{podName}, framework.PodStartTimeout) {
```

Comment:
👍 Though it doesn't seem strictly necessary for this change, your version of this is much easier to understand. I had to go lookup what `waitForOrFailWithDebug(1)` did and how `forEachPod` was working.